community/kernel/src/main/java/org/neo4j/kernel/api/security/AuthSubject.java

```diff
@@ -54,6 +54,22 @@
     String username();
 
     /**
+     * @param username a username
+     * @return true if the provided username is the underlying user name of this subject
+     */
+    boolean hasUsername( String username );
+
+    /**
+     * Ensure that the provided username is the name of an existing user known to the system.
+     *
+     * @param username a username
+     * @throws InvalidArgumentsException if the provided user name is not the name of an existing user
+     */
+    default void ensureUserExistsWithName( String username ) throws InvalidArgumentsException {
```

Comment:
This is unrelated to this PR. We're just moving code around here, I'll try to factor our the default implementation. This whole hierarchy needs some overhaul but out of scope for this PR.